Allow to choose Dark/Light tray icon

hron opened the pull request:
Some quick demo:


https://github.com/user-attachments/assets/183c3d54-a534-4268-94a2-77922941eda5



Closes #1356

hron committed:
Allow to choose Dark/Light tray icon

Closes #1356

Alipoodle commented:
_this is without testing and checking docs_
But trayTemplate was a specific name required for MacOS to automatically do it on that OS~
So this maybe something which breaks that functionality ~

hron committed:
Unify light/dark tray icon switching

Before this change, the app displayed a dark tray icon on Linux and a
red one on Windows, regardless of the OS's dark mode setting. On macOS,
it used the OS mechanism via tray image names *Template.png, allowing
macOS to automatically invert the tray icon colors.

With this change, the app now relies on the Electron API: nativeTheme.
This works on macOS, Linux, and Windows. When the OS 'Dark Mode' setting
is enabled, the app will choose a white tray icon; otherwise, it will
use a black one.

Additionally, there is a new setting that allows the user to force the
tray icon to be either black or white. This is useful in situations
where a light theme is in use, but the tray panel is still black, which
is the default configuration on Ubuntu, for example.

hron commented:
@Alipoodle Thank you for the quick feedback! I trust you about macOS, but I think I have found a universal way to handle light/dark tray icon style which can be utilized everywhere.  https://www.electronjs.org/docs/latest/api/native-theme 

I cannot test it on macOS, but it works on Linux. The new version goes one step further and introduces a setting to either do light/dark switching automatically depending on 'Dark Mode' setting or allow the user to choose. This is useful for situation when a light theme is in use, but the panel with tray is dark. 

The updated version should work on Windows as well as you requested. Please let me know if something is missing. Also, here is new demo:

https://github.com/user-attachments/assets/672c4e47-6a98-4c96-81b8-f51b9a0ffed5

NovusTheory commented:
> Additionally this feels like it would be nice to also include Windows and have a Inverted .ico file as well.

The normal for Windows is usually to just use the app's colored icon and not some specialized tray icon which can flip colors like this.

While this PR is only to toggle dark/light icons I feel like if we'll be messing with the tray icon we should just try and swap to something better like maybe utilizing our black and white logos that are already in the repo (we already use the white for the titlebar). Mac is still a special case though as UI and designs there are drastically different

hron committed:
Use black and white logos for tray icon on Linux

hron committed:
Use the colored tray icon on Windows

hron committed:
Use trayTemplate.png as tray icon on macOS

hron committed:
Remove unnecessary tray icon files

hron committed:
Show the tray icon style setting only on Linux

hron commented:
@NovusTheory all your feedback has been incorporated. In addition, the setting to change the tray icon style is only shown on Linux now. I think it makes sense because on macOS the app uses trayTemplate.png and on Windows it always shows the colored version of the tray icon, so the setting doesn't make any sense there.

NovusTheory commented:
Outside of the CSS for the select menu missing and not aligning with the rest of the app's style I'm satisfied with these changes.

hron committed:
Add CSS for "Tray icon style" setting

hron commented:
Well, CSS is not my strong side, but here are some improvements:


https://github.com/user-attachments/assets/e30d5ee6-c518-4be1-9745-a41c0934ca0e

NovusTheory commented:
I had just got reminded of an old branch I made for something else which implemented a select menu setting https://github.com/ytmdesktop/ytmdesktop/commit/363916b3418e1e30b095ef2edc6aeb2118bc246b. After looking over the code for this PR I think I'd prefer the logic in the linked commit for our setting component. I can cherry-pick that select menu code for our setting component (YTMDSetting.vue) into the development branch and this PR could get adjusted to utilize that if you'd be willing to. If that branch had ever gotten merged then it would've already been available and ready to use for this PR.

NovusTheory commented:
@hron Pushed the select menu type to development and ready for use.

hron committed:
Use `select` component for the tray icon style

hron commented:
@NovusTheory okay, it's done and ready for another review round

NovusTheory commented on src/renderer/windows/settings/Settings.vue:
```diff
@@ -313,6 +326,14 @@ window.ytmd.handleUpdateDownloaded(() => {
+          <YTMDSetting
+            v-if="isLinux"
+            v-model="trayIconStyle"
+            :options-map="toOptionsMap(trayIconStyles)"
```
Nit: This should capitalize the options for the UI. The actual config doesn't need it.

NovusTheory reviewed (changes requested):
Not sure if it would be beneficial to make the config an enum (number) instead of arbitrary text but otherwise seems okay outside a nit change.

@Alipoodle Chiming you in for thoughts on the above.

hron committed:
Use numbers for `trayIconStyle` in config

hron commented:
@NovusTheory any more comments on this? I think the PR is ready for merging

NovusTheory reviewed (approved):
Sorry this took so long to get around to. LGTM
